packages/authentication/src/types.ts

```diff
@@ -110,3 +138,30 @@ export function registerAuthenticationStrategy(
     },
   );
 }
+
+/**
+ * Registers a security scheme spec as an extension of the @loopback/openapi-v3
+ * OAS_ENHANCER_EXTENSION_POINT_NAME extension point.
+ *
+ * @param context - Context object
+ * @param securityScheme - security scheme object
+ */
+export function registerAuthSecuritySpec(
+  context: Context,
+  securityScheme: SecurityScheme,
+) {
+  @bind(asSpecEnhancer)
+  class SecuritySpecEnhancer implements OASEnhancer {
```

Comment:
I have mixed feelings about creating a new `SecuritySpecEnhancer` class for each authentication strategy registered. Have you considered an alternative approach where each `AuthenticationStrategy` class can decide to implement `OASEnhancer` and register itself `asSpecEnhancer`?

IMO, such solution will require less code in the framework and offer more flexibility to strategy authors.

A mock-up:

```ts
import {
   asSpecEnhancer,
   // new helper API to be implemented
   mergeSecuritySchemeToSpec,
   OASEnhancer,
 } from '@loopback/openapi-v3';

@bind(
  asSpecEnhancer,
 // should we also add something like asAuthStrategy
 // to correctly set the binding tags needed to annotate
 // this class as an authentication strategy?
)
export class JWTAuthenticationStrategy implements AuthenticationStrategy, OASEnhancer {
  // ...

  modifySpec(spec: OpenApiSpec): OpenApiSpec {
    return mergeSecuritySchemeToSpec(spec, 'jwt', {
      type: 'http',
      scheme: 'bearer',
      bearerFormat: 'JWT',
    });
  }
}

// inside @loopback/openapi-v3'

export function mergeSecuritySchemeToSpec(spec: OpenApiSpec, schemeName: string, schemeSpec: SecurityScheme) {
  const patchSpec = {
    components: {
      securitySchemes: {[schemeName]: schemeSpec},
    },
  };

  const mergedSpec = mergeOpenAPISpec(spec, patchSpec);
  return mergedSpec;
}
```

Regardless of the approach we choose for integrating AuthenticationSchemes and OASEnhancer, I think we should always extract the helper `mergeSecuritySchemeToSpec` to `@loopback/openapi-v3`, because it's something that can be used in different places too. (Feel free to pick a better function name.)

Reply:
@bajtos Much better your way! 👌 I've created `asAuthStrategy` but couldn't figure out how to do something like `@bind(asSpecEnhancer, asAuthStrategy)`.  The documents seem to cover a single binding but not multiple.

Thank you for your guidance, it is most appreciated!

Reply:
The signature of `@bind` is `export function bind(...specs: BindingSpec[])`. It can take any number of binding template functions, such as `@bind(asSpecEnhancer, asAuthStrategy)`.

Reply:
I did try `@bind(asSpecEnhancer, asAuthStrategy)` but are there any considerations when adding to context?  I used only the following: `context.add(createBindingFromClass(strategyClass));` and found that only one of the bindings was happy depending on the order in the `@bind` decorator.  I ran out of time to poke at it, is it something simple I need to do?  Thank you for your help.

Reply:
See https://github.com/strongloop/loopback-next/blob/master/packages/context/src/__tests__/unit/binding-decorator.unit.ts#L96

Reply:
It seems that you hit the limitation of https://github.com/strongloop/loopback-next/blob/master/packages/core/src/extension-point.ts#L127. The `extensionFor` uses the same tag to point to an `extensionPoint` and that's why two templates override each other.

Reply:
What about https://github.com/strongloop/loopback-next/pull/4816?

Reply:
#4816 is now merged into master. Please rebase to pick it up.